documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+| MQTT with Windows Certificate Store Method                             | $${\Large\color{red}&#10008}$$        | $${\Large\color{red}&#10008}$$        | $${\Large\color{green}&#10004}$$ | $${\Large\color{red}&#10008}$$   | TODO Node.js section |
+| MQTT with PKCS11 Method                                                | $${\Large\color{red}&#10008}$$        | $${\Large\color{red}&#10008}$$        | $${\Large\color{green}&#10004}$$ | $${\Large\color{red}&#10008}$$   | [Node.js section](https://github.com/awslabs/aws-crt-nodejs/blob/main/MQTT5-UserGuide.md#direct-mqtt-with-pkcs11-method) |
+| HTTP Proxy                                                             | $${\Large\color{orange}&#10004\*\*}$$ | $${\Large\color{orange}&#10004\*\*}$$ | $${\Large\color{green}&#10004}$$ | $${\Large\color{green}&#10004}$$ | [Node.js section](https://github.com/awslabs/aws-crt-nodejs/blob/main/MQTT5-UserGuide.md#http-proxy) [Browser section](https://github.com/awslabs/aws-crt-nodejs/blob/main/MQTT5-UserGuide.md#http-proxy-1) |
+
+${\Large\color{orange}&#10004\*}$ - To get this connection type work in the v1 SDK, you need to extract a private key from a PKCS #12 file.\
+${\Large\color{orange}&#10004\*\*}$ - The v1 SDK does not allow specifying HTTP proxy, but it is possible to configure systemwide proxy.
+
+#### Example of creating connection in the v1 SDK
+
+```typescript
+var awsIot = require('aws-iot-device-sdk');
+
+var clientEndpoint = "<prefix>-ats.iot.<region>.amazonaws.com";
+var clientId = "<unique client id>";
+var certificateFile = "<certificate file>";  // X.509 based certificate file
+var privateKeyFile = "<private key file>";   // PEM encoded private key file
+
+// A connection will be established on instantiating a device object.
+var device = awsIot.device({
+   keyPath: privateKeyFile,
+  certPath: certificateFile,
+  clientId: clientId,
+      host: clientEndpoint
+});
+```
+
+#### Example of creating connection in the v2 SDK
+
+```typescript
+import { mqtt5, iot } from "aws-iot-device-sdk-v2";
+
+const clientEndpoint : string = "<prefix>-ats.iot.<region>.amazonaws.com";
+const clientId : string = "<unique client id>";
+const certificateFile : string = "<certificate file>";  // X.509 based certificate file
+const privateKeyFile : string = "<private key file>";   // PEM encoded private key file
+
+let builder = iot.AwsIotMqtt5ClientConfigBuilder.newDirectMqttBuilderWithMtlsFromPath(
+        clientEndpoint,
+        certificateFile,
+        privateKeyFile
+
+builder.withConnectProperties({
+    clientId: clientId
+});
+
+let config : mqtt5.Mqtt5ClientConfig = builder.build();
+
+let client : mqtt5.Mqtt5Client = new mqtt5.Mqtt5Client(config);
+
+// Connection must be started explicitly.
```

Comment:
Lets frame this as a positive. v1 has to initiale connection as soon as a client is created but v2 client can be set to connect when user wants it to be. e.g.  "Connection of client is controlled and started explicitly with start()."